### ISSUE TITLE
PowerShell Universal Git Synchronization Fails with Authentication and "Too Many Redirects" Error – Version 5.0.9

### ISSUE THREAD

**Issue by RubenT91** [ironmansoftware/powershell-universal#3904]
### Version

5.0.9

### Severity

Critical

### Environment

msi

### Steps to Reproduce

- Configure Git synchronization on 14 PSU servers.
- Attempt to push changes via PSU’s Git sync functionality using the configured token authentication.
- Observe the error: "too many redirects" and authentication replays, preventing sync completion.
- Trevor’s team was able to use Git directly (outside PSU) to authenticate and clone the repository using a personal access token.
- They provided examples of API-based access via Postman with Bearer tokens working, but HTTP(S) access (which PSU seems to rely on) fails.

### Expected behavior

```console
PSU should authenticate with the provided GitLab personal access token, upload changes, and pull down updates to each PSU server according to the interval specified in PSU's Git settings.
```

### Actual behavior

```console
PSU fails to authenticate with GitLab. It shows errors like "too many redirects" and "authentication replays." Trevor’s team could successfully authenticate and clone the repository outside PSU using Git, confirming that GitLab is accessible.
```

### Additional Environment data

Operating System: All servers are running Windows Server 2019 with only PSU installed.
GitLab Instance: On-premise, no cloud-based repositories.
Token Configuration: The user uses a personal access token with read/write permissions to the GitLab repository.

**Comment by JessePeden** [comment]
This issue has been around since at least 4.0.x and [for me] sporadically happens during overnight peak traffic times for any repository I've tested.  The issue goes away once traffic dies down on the internet.  I'm guessing that the issue stems from people doing most of their heavy backup jobs and syncs during early morning hours or late at night.

**Comment by RubenT91** [comment]
A solution to this case was found thanks to a comment from a PSU user, for which this ticket was opened:

https://docs.powershelluniversal.com/config/git 

In that link is a section where it showed I can update the Username and Password used for Git. Within the appsettings.json file, I added (highlighted in yellow):

    "Data": {

    "RepositoryPath": "%ProgramData%\\UniversalAutomation\\Repository",

    "ConnectionString": "%ProgramData%\\UniversalAutomation\\database.db",

    "GitRemote": https://mypathtogitlab/myrepo.git,

    "GitUserName": "Private-Token",

    "GitPassword": "my_actual_token"

  },

 

Then I restarted PSU and immediately the Git errors stopped and a Git sync was successful. So, this was, as I expected, a simple configuration change to have authentication work.

NOTE: Previous to a recent GitLab update, GitLab accepted any Username and if the password matched a valid token for the repository, it used it. GitLab, after the recent update to our environment, no longer allows any username when authenticating via http(s).

For us, after the update, GitLab will only accept: “Private-Token” as a username and the Token must be valid with the correct permissions for the repository.

 

So, PSU is now successfully syncing with GotLab. I am not sure why you opened a case with GitHub. This was always getting the authentication settings correct.
